### ISSUE TITLE
dns-horizontal-autoscaler missing ServiceAccount

### ISSUE THREAD

**Issue by justinsb** [kubernetes/kubernetes#43496]
I note that there is no serviceaccount on dns-horizontal-autoscaler.

https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns-horizontal-autoscaler/dns-horizontal-autoscaler.yaml

It does not appear to work under (my) RBAC configuration therefore.

Do we know what permissions it needs?  Certainly list nodes....

**Comment by justinsb** [comment]
Here's what I've got so far...


I put `serviceAccountName: kube-dns-autoscaler` on the pod

Then I apply the manifest below.

A few things I'm unsure about:

* Do we need any other permissions for kube-dns-autoscaler?  And I didn't see `update` configmaps, but I'm guessing if we create it that we also update it?
* Is there any way to scope the configmaps?   I guess I could create a Role for the configmaps & deployments to bind them to the kube-system namespace, but it would be good IMO also to bind them to the specific deployment & configmap that we know to be the target.

```

apiVersion: v1
kind: ServiceAccount
metadata:
  name: kube-dns-autoscaler
  namespace: kube-system
  labels:
    k8s-addon: kube-dns.addons.k8s.io

---

apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRole
metadata:
  labels:
    k8s-addon: kube-dns.addons.k8s.io
  name: kube-dns-autoscaler
rules:
- apiGroups:
  - ""
  resources:
  - nodes
  verbs:
  - list
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - get
  - create
  - update
- apiGroups:
  - "extensions"
  resources:
  - deployments/scale
  verbs:
  - get
  - update

---

apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  labels:
    k8s-addon: kube-dns.addons.k8s.io
  name: kube-dns-autoscaler
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: kube-dns-autoscaler
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: User
  name: system:serviceaccount:kube-system:kube-dns-autoscaler


```

**Comment by ncdc** [comment]
cc @liggitt

**Comment by liggitt** [comment]
almost all add-ons run as the `default` service account in the `kube-system` namespace. if a deployment wants to make use of add-ons that don't yet use their own service account or provide a custom role, a short-term solution is to give the default service account in kube-system superuser permissions

**Comment by bgrant0607** [comment]
cc @MrHohn

**Comment by MrHohn** [comment]
There is also an open PR for this: #42617

**Comment by MrHohn** [comment]
Note that ConfigMap permission may be tweaked once we mount configmap into dns-autoscaler pod.

**Comment by roberthbailey** [comment]
Looks like this was fixed by #42617. Please re-open if there is more to do.
